plone/app/layout/viewlets/common.py

```diff
@@ -271,6 +272,13 @@ def current_language(self):
             or self.language_settings.default_language
         )
 
+    @property
+    @memoize_forever
```

Comment:
Please note the memoize_forever. I think this can be memoized forever, but if anyone changes the registry entry it would be strange if the change doesn't reflect immediately.
Opinions?

Reply:
Pay attention that is module is not thread-safe, so use it with care.

These remember a value "forever", i.e. until the process is restarted. They work on both global functions and class functions.:

"Pay attention that is module is not thread-safe, so use it with care.

These remember a value "forever", i.e. until the process is restarted. They work on both global functions and class functions.:"

Isn't better just to cache the value from the registry itself in the registry adapter? So you invalidate the cache when the value is set again. It is impossibile to cache a list because basically the key is the list itself. A "cache me" checkbox could be added to the registry entries, so you can decide which record entries to cache in memory.

Reply:
> Isn't better just to cache the value from the registry itself in the registry adapter? So you invalidate the cache when the value is set again. It is impossibile to cache a list because basically the key is the list itself. A "cache me" checkbox could be added to the registry entries, so you can decide which record entries to cache in memory.

This would be a great feature for plone.registry. But its out of scope here.

@thet it would be enough to cache the value per request, not forever, this might avoid side effects. same for master.

Reply:
Thans for the pointers @yurj @jensens !
I now use ``@memoize_contextless``.

Reply:
> 

Per request is useful if you call it many times but it is called once. @thet is caching forever this values for the navigation viewlet, which means you've to restart the instance after changing the value in the control panel. This apply only to navigation, while listing get it correctly from registry.

Are registry so terrible in performance when getting a list? It basically uses OOBTree with some logic for the name if it is an interface (but this should cached efficently by z3c).

Reply:
yes, probably caching is pointless here.